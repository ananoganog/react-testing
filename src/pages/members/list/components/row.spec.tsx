import React from 'react';
import { shallow } from 'enzyme';
import { Row } from './row';
import { Member } from '../viewModel';


describe('pages/members/list/components spec', () => {
  describe('Row', () => {
    it('should render a row with 3 td when passing a member', () => {
      // Arrange
      const member: Member = {
        id: 0,
        name: 'someone',
        avatarUrl: 'someone\'s avatar url',
      };

      // Act
      const component = shallow(<Row member={member} />)

      // Assert
      expect(component).toMatchSnapshot();
    });
  });
});
