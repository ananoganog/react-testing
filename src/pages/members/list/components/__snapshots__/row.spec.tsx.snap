// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/members/list/components spec Row should render a row with 3 td when passing a member 1`] = `
<tr>
  <td>
    <img
      src="someone's avatar url"
      style={
        Object {
          "maxWidth": "10rem",
        }
      }
    />
  </td>
  <td>
    <span>
      0
    </span>
  </td>
  <td>
    <span>
      someone
    </span>
  </td>
</tr>
`;
